src/components/plots/barPlot.ts

```diff
@@ -151,6 +166,114 @@ export module Plot {
     }
 
     /**
+     * Retrieves the closest PlotData to queryPoint.
+     *
+     * Bars containing the queryPoint are considered closest. If queryPoint lies outside
+     * of all bars, we return the closest in the dominant direction (x for horizontal
+     * charts, y for vertical) and break ties
+     *
+     * @param {Point} queryPoint The point to which dataset points should be compared
+     *
+     * @returns {PlotData} The PlotData closest to queryPoint
+     */
+    public getClosestPlotData(queryPoint: Point): PlotData {
+      var keys = this.datasetOrder();
+      var chartXExtent = { min: 0, max: this.width() };
+      var chartYExtent = { min: 0, max: this.height() };
+
+      var minDist = Infinity;
+      var closest: any;
+
+      var accessor: { [axis: string]: (p: Point) => number } = {
+        "x": (p) => p.x,
+        "y": (p) => p.y
+      };
+
+      var dominantAxis: string = this._isVertical ? "x" : "y";
+      var secondaryAxis: string = this._isVertical ? "y" : "x";
+
+      keys.forEach((key) => {
+        var plotData = this.getAllPlotData(key);
+        if (plotData != null && plotData.pixelPoints != null) {
+          plotData.pixelPoints.forEach((plotPt, i) => {
+            var bar = plotData.selection[0][i];
+
+            if (!Bar._intersectsBBox(chartXExtent, chartYExtent, bar.getBBox())) {
+              // bar isn't visible on plot; ignore it
+              return;
+            }
```

Comment:
We don't currently have a function for bar visibility.  I think this function should be built out rather than using _intersectsBBox to proxy for it.